lib/opentelemetry/trace/exporter/http_client.lua

```diff
@@ -28,29 +33,52 @@ function _M.new(address, timeout, headers)
         uri = uri,
         timeout = timeout,
         headers = headers,
+        httpc = httpc,
```

Comment:
You can do `httpc or http.new()` here to clean things up IMO

Reply:
More importantly, if I reload openresty, I get the following error from the `init_worker_by_lua` block:

```
openresty_1       | 2024/02/07 02:05:43 [error] 257#257: init_worker_by_lua error: /usr/local/openresty/luajit/share/lua/5.1/resty/http.lua:133: API disabled in the context of init_worker_by_lua*
openresty_1       | stack traceback:
openresty_1       | 	[C]: in function 'ngx_socket_tcp'
openresty_1       | 	/usr/local/openresty/luajit/share/lua/5.1/resty/http.lua:133: in function 'new'
openresty_1       | 	...are/lua/5.1/opentelemetry/trace/exporter/http_client.lua:36: in function 'exporter_client_new'
openresty_1       | 	init_worker_by_lua:10: in main chunk
```

Reply:
TL;DR: my understanding if this error message is that `http.new` must be called within a request, not during the worker initialization

Reply:
case and point; this is how it was done before https://github.com/yangxikun/opentelemetry-lua/pull/87/files#diff-98945ad3dcad8721c3a0663c81282927bc026ec2addb95b39a0f592c60de2176L36

My goal with making this injectable was to make it easier to test, so that we can `spy` on the instance's function calls

Reply:
Makes sense, thanks for the explanation 🎉